mlflow/utils/cli_args.py

```diff
@@ -137,6 +137,10 @@ def _resolve_env_manager(ctx, _, env_manager):
 
 PORT = click.option("--port", "-p", default=5000, help="The port to listen on (default: 5000).")
 
+TIMEOUT = click.option(
+    "--timeout", "-t", default=60, help="Timeout in seconds to serve a request (default: 60)."
```

Comment:
It looks like `RFuncBackend.serve` failed:

https://github.com/mlflow/mlflow/runs/6221444918?check_suite_focus=true

Can we set the default to `None`?

Reply:
@harupy Imo the failed test is not due timeout.
A similar error occurred to already merged PRs such [PR1](https://github.com/mlflow/mlflow/runs/6146781571?check_suite_focus=true) or [PR2](https://github.com/mlflow/mlflow/runs/6166017339?check_suite_focus=true)
Also does this error relate to this [fix](https://github.com/mlflow/mlflow/pull/5763) you pushed?

Reply:
I got this error by serving a model logged in R:

```
$ mlflow models serve --model-uri runs:/5f29b8431ec74ec6bfd8e58378a8a757/model
Traceback (most recent call last):
  File "/home/haru/miniconda3/envs/mlflow-dev-env/bin/mlflow", line 33, in <module>
    sys.exit(load_entry_point('mlflow', 'console_scripts', 'mlflow')())
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/haru/Desktop/repositories/mlflow/mlflow/models/cli.py", line 70, in serve
    model_uri=model_uri, port=port, host=host, timeout=timeout, enable_mlserver=enable_mlserver
  File "/home/haru/Desktop/repositories/mlflow/mlflow/rfunc/backend.py", line 62, in serve
    raise Exception("Timeout is not yet supported in the R backend.")
Exception: Timeout is not yet supported in the R backend.
```

Reply:
> A similar error occurred to already merged PRs such [PR1](https://github.com/mlflow/mlflow/runs/6146781571?check_suite_focus=true) or [PR2](https://github.com/mlflow/mlflow/runs/6166017339?check_suite_focus=true)

Yes. https://github.com/mlflow/mlflow/pull/5763 fixed the issue.

Reply:
> @sniafas
> 
> It looks like `RFuncBackend.serve` failed:
> 
> https://github.com/mlflow/mlflow/runs/6221444918?check_suite_focus=true
> 
> Can we set the default value to `None` here and set `timeout` to `60` in `mlflow.pyfunc.scoring_server.get_cmd` if it's `None`?

It's a nice idea, but setting None as default would passed to scoring server no matter what.

Reply:
> setting None as default would passed to scoring server no matter what.

Are there any downsides if none is passed to scoring server?

Reply:
Let's use `default=60` and mention backends that don't support this option.

Reply:
> > setting None as default would passed to scoring server no matter what.
> 
> Are there any downsides if none is passed to scoring server?

The downside is that ``mlflow models serve -m "model" will not run and `timeout` arg is always needed. Since timeout a new argument, will always get through backend right ?

Reply:
> timeout arg is always needed.

If we set `required=True`.

Reply:
One advantage is we can thrown an exception when a user tries to use timeout in a backemd that doesn't support it.